src/lib.rs

```diff
@@ -22,104 +63,192 @@ impl fmt::Display for DeviceType {
+const VENDOR_ID: u16 = 0x046d;
+const USAGE_PAGE: u16 = 0xff43;
+
+fn device_type_from_product_id(product_id: u16) -> Option<DeviceType> {
+    match product_id {
+        0xc900 => DeviceType::LitraGlow.into(),
+        0xc901 => DeviceType::LitraBeam.into(),
+        0xb901 => DeviceType::LitraBeam.into(),
+        0xc903 => DeviceType::LitraBeamLX.into(),
+        _ => None,
     }
 }
 
 const MINIMUM_TEMPERATURE_IN_KELVIN: u16 = 2700;
 const MAXIMUM_TEMPERATURE_IN_KELVIN: u16 = 6500;
 
-pub fn get_connected_devices<'a>(
-    api: &'a HidApi,
-    serial_number: Option<&'a str>,
-) -> impl Iterator<Item = Device> + 'a {
-    let litra_devices = api
-        .device_list()
-        .filter(|device| {
-            device.vendor_id() == VENDOR_ID
-                && PRODUCT_IDS.contains(&device.product_id())
-                && device.usage_page() == USAGE_PAGE
-        })
-        .filter(move |device| {
-            serial_number.is_none()
-                || serial_number.as_ref().is_some_and(|expected| {
-                    device
-                        .serial_number()
-                        .is_some_and(|actual| &actual == expected)
-                })
-        });
-
-    litra_devices
-        .filter_map(|device| match api.open_path(device.path()) {
-            Ok(device_handle) => Some((device, device_handle)),
-            Err(err) => {
-                println!("Failed to open device {:?}: {:?}", device.path(), err);
-                None
-            }
-        })
-        .map(|(device, device_handle)| {
-            let device_type = get_device_type(device.product_id());
-            let is_on = is_on(&device_handle, &device_type);
-            let brightness_in_lumen = get_brightness_in_lumen(&device_handle, &device_type);
-            let temperature_in_kelvin = get_temperature_in_kelvin(&device_handle, &device_type);
-            let minimum_brightness_in_lumen = get_minimum_brightness_in_lumen(&device_type);
-            let maximum_brightness_in_lumen = get_maximum_brightness_in_lumen(&device_type);
-
-            Device {
-                serial_number: device.serial_number().unwrap_or("").to_string(),
-                device_type,
-                is_on,
-                brightness_in_lumen,
-                temperature_in_kelvin,
-                device_handle,
-                minimum_brightness_in_lumen,
-                maximum_brightness_in_lumen,
-                minimum_temperature_in_kelvin: MINIMUM_TEMPERATURE_IN_KELVIN,
-                maximum_temperature_in_kelvin: MAXIMUM_TEMPERATURE_IN_KELVIN,
-            }
-        })
+/// Returns an [`Iterator`] of connected devices supported by this library.
+pub fn get_connected_devices(api: &HidApi) -> impl Iterator<Item = Device<'_>> {
+    api.device_list()
+        .filter_map(|device_info| Device::try_from(device_info).ok())
 }
 
-fn generate_is_on_bytes(device_type: &DeviceType) -> [u8; 20] {
+fn generate_is_enabled_bytes(device_type: &DeviceType) -> [u8; 20] {
```

Comment:
I changed it because there were two different functions to generate bytes for setting the state even though the only differ in one byte. And I found `generate_turn_on_bytes` misleading if the same function can also be used to generate turn off bytes.

I adapted the other function names for consistency. How about `is_powered`/`set_powered`? Or `is_on`/`set_on`? We can of course add convenience methods like `turn_on` that just call `set_powered(true)` internally if you want.

Reply:
Can we go for `is_on`/`set_on` please? With that change, I'd be happy to merge.